common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -213,16 +240,18 @@ public static String toHexString(byte[] src, int offset, int length) {
     public static int decodeHexNibble(final char c) {
         // Character.digit() is not used here, as it addresses a larger
         // set of characters (both ASCII and full-width latin letters).
-        if (c >= '0' && c <= '9') {
-            return c - '0';
-        }
-        if (c >= 'A' && c <= 'F') {
-            return c - ('A' - 0xA);
-        }
-        if (c >= 'a' && c <= 'f') {
-            return c - ('a' - 0xA);
+        final int index = c;
+        assert index >= 0;
```

Comment:
redundant by spec:
"char: The char data type is a single 16-bit Unicode character. It has a minimum value of '\u0000' (or 0) and a maximum value of '\uffff' (or 65,535 inclusive)."

https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html